config/samples/_v1_agentaction.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: porter.sh/v1
+kind: AgentAction
+metadata:
+  name: agentaction-sample
+spec:
+  args: ["installation", "apply", "installation.yaml"]
+  files:
+    # base64 encoded file contents
+    installation.yaml: c2NoZW1hVmVyc2lvbjogMS4wLjAKbmFtZXNwYWNlOiBvcGVyYXRvcgpuYW1lOiBoZWxsbwpidW5kbGU6CiAgcmVwb3NpdG9yeTogZ2hjci5pby9nZXRwb3J0ZXIvdGVzdC9wb3J0ZXItaGVsbG8KICB2ZXJzaW9uOiAwLjIuMApwYXJhbWV0ZXJzOgogIG5hbWU6IGxsYW1hcyAK
```

Comment:
You can always delete a resource by editing it and removing any finalizers on it.

I went with base64 because it allows people to add any files they need into the working directory of the agent, without us having known about it up-front. Do you have a suggestion for how you'd like to see it done differently while still being flexible?

Reply:
I am concerned that it may be an indentation yaml bug generator because it isn't embedding a yaml document, it's a formatted yaml string embeded in a yaml file. Though either way it makes it much easier to introduce bugs in the document.

Since k8s users are already familiar with how to decode secrets, I am not too worried about the UX of doing exactly what secrets do for this. Let's go with following k8s existing conventions for now and if we need to revisit in later versions, I'm fine with that. I'd rather get this merged so we can try out a working operator and move forward with more controllers.